DQMOffline/Trigger/plugins/METMonitor.cc

```diff
@@ -0,0 +1,276 @@
+}
+
+void METMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+				 edm::Run const        & iRun,
+				 edm::EventSetup const & iSetup) 
+{  
+  
+  std::string histname, histtitle;
+
+  std::string currentFolder = folderName_ ;
+  ibooker.setCurrentFolder(currentFolder.c_str());
+
+  histname = "met"; histtitle = "PFMET";
+  bookME(ibooker,metME_,histname,histtitle,met_binning_.nbins,met_binning_.xmin, met_binning_.xmax);
+  setMETitle(metME_,"PF MET [GeV]","events / [GeV]");
+
+  histname = "met_variable"; histtitle = "PFMET";
+  bookME(ibooker,metME_variableBinning_,histname,histtitle,met_variable_binning_);
+  setMETitle(metME_variableBinning_,"PF MET [GeV]","events / [GeV]");
+
+  histname = "metVsLS"; histtitle = "PFMET vs LS";
+  bookME(ibooker,metVsLS_,histname,histtitle,ls_binning_.nbins, ls_binning_.xmin, ls_binning_.xmax,met_binning_.xmin, met_binning_.xmax);
+  setMETitle(metVsLS_,"LS","PF MET [GeV]");
+
+  histname = "metPhi"; histtitle = "PFMET phi";
+  bookME(ibooker,metPhiME_,histname,histtitle, phi_binning_.nbins, phi_binning_.xmin, phi_binning_.xmax);
+  setMETitle(metPhiME_,"PF MET #phi","events / 0.1 rad");
+
+  // Initialize the GenericTriggerEventFlag
+  if ( num_genTriggerEventFlag_ && num_genTriggerEventFlag_->on() ) num_genTriggerEventFlag_->initRun( iRun, iSetup );
+  if ( den_genTriggerEventFlag_ && den_genTriggerEventFlag_->on() ) den_genTriggerEventFlag_->initRun( iRun, iSetup );
+
+}
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+void METMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)  {
+
+  // Filter out events if Trigger Filtering is requested
+  if (den_genTriggerEventFlag_->on() && ! den_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
+
+  edm::Handle<reco::PFMETCollection> metHandle;
+  iEvent.getByToken( metToken_, metHandle );
+  reco::PFMET pfmet = metHandle->front();
+  if ( ! metSelection_( pfmet ) ) return;
+  
+  float met = pfmet.pt();
+  float phi = pfmet.phi();
+
+  edm::Handle<reco::PFJetCollection> jetHandle;
+  iEvent.getByToken( jetToken_, jetHandle );
+  std::vector<reco::PFJet> jets;
+  for ( auto j : *jetHandle ) {
+    if ( jetSelection_( j ) ) jets.push_back(j);
+  }
+  if ( int(jets.size()) < njets_ ) return;
+  
+  edm::Handle<reco::GsfElectronCollection> eleHandle;
+  iEvent.getByToken( eleToken_, eleHandle );
+  std::vector<reco::GsfElectron> electrons;
+  for ( auto e : *eleHandle ) {
+    if ( eleSelection_( e ) ) electrons.push_back(e);
+  }
+  if ( int(electrons.size()) < nelectrons_ ) return;
+  
+  edm::Handle<reco::MuonCollection> muoHandle;
+  iEvent.getByToken( muoToken_, muoHandle );
+  std::vector<reco::Muon> muons;
+  for ( auto m : *muoHandle ) {
+    if ( muoSelection_( m ) ) muons.push_back(m);
+  }
+  if ( int(muons.size()) < nmuons_ ) return;
```

Comment:
is it worth while to check also before running the selection that the number of electrons/muons can pass this same threshold? This might save a lot of checking of the selectors